src/electron-main.js

```diff
@@ -596,8 +602,15 @@ ipcMain.on('seshat', async function(ev, payload) {
                         // index.
                         await eventIndex.changePassphrase(newPassphrase);
 
+                        // Keep this delay to avoid race conditions where the DB
+                        // lock has not been released properly when trying to
+                        // create a new event index
+                        await delay(500);
```

Comment:
I am a bit wary of just papering over a race with a delay like this... On fast systems, it might be wasted time, and on slow systems, it might not be enough time.

Could you work with @poljar to see how we can adjust Seshat to fix this without needing delays at the application level? I suspect they'll have some good ideas on how the Seshat side can be improved, and possibly time to cook up a PR for it as well.

Reply:
Is that race a real concern? As far as I remember the `changePassphrase()` method consumes and thus drops the database on the Rust side. Perhaps windows behaves somewhat differently?

Reply:
Below is a screenshot of the error message I encounter when removing my hack `delay` function call

![Screen Shot 2021-04-21 at 13 51 08](https://user-images.githubusercontent.com/769871/115556541-b502d800-a2a8-11eb-99c1-5162dde6bdb9.png)

Reply:
Seems that we don't call the right method after the passphrase has been changed: https://github.com/matrix-org/seshat/pull/85

Reply:
Awesome, that should remove the need for my delay then!
I'll get rid of that bit and try to compile a local version with your fix to confirm that it all works smoothly

Reply:
> Would it work to generate a random passphrase straight away? Why do we need to create a db with a default passphrase only to change it seconds after?

If I remember correctly that code upgrades users from the default passphrase to the random one.

Reply:
I have tried applying your changes locally and removed the hacky delay I added. 

I'm still prompted with that lock failure I was experiencing before